### ISSUE TITLE
Deadlock in PostgreSQL 12

### ISSUE THREAD

**Issue by romash1408** [flyway/flyway#2634]
Flyway version: **6.1.4 (API)**
DB version: **PostgreSQL 12.1 on x86_64-pc-linux-gnu, compiled by gcc (GCC) 9.2.0, 64-bit**

This code is calling flyway on start up:
```java
Flyway flyway = Flyway.configure()
        .baselineOnMigrate(true)
        .dataSource(dataSource())
        .load();
flyway.migrate();
```
Then flyway tries to lock table, but it fails on this line
https://github.com/flyway/flyway/blob/72cfe7db4a21a44001648ea8e00aef70849cf24d/flyway-core/src/main/java/org/flywaydb/core/internal/database/postgresql/PostgreSQLAdvisoryLockTemplate.java#L112
because `rs.getString("pg_try_advisory_lock")` returns `true`, so we've got infinite loop

**Comment by juliahayward** [comment]
This is somewhat unexpected as the documentation claims Postgres always emits `t` or `f` - see https://www.postgresql.org/docs/current/datatype-boolean.html - however your PR makes the code better in any case. Thankyou!

**Comment by romash1408** [comment]
Potential fix #2635

**Comment by juliahayward** [comment]
Possibly related to #2506.
